packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -180,27 +314,69 @@ class DatePickerInput extends React.Component {
     const buttonText = intl.formatMessage({ id: 'Terra.datePicker.openCalendar' });
 
     return (
-      <div className={cx('custom-input')}>
-        <input
-          // Create a hidden input for storing the name and value attributes to use when submitting the form.
-          // The data stored in the value attribute will be the visible date in the date input but in ISO 8601 format.
-          data-terra-date-input-hidden
-          type="hidden"
-          name={name}
-          value={dateValue}
-        />
-        <Input
-          {...additionalInputProps}
-          className={inputClasses}
-          type="text"
-          name={'terra-date-'.concat(name)}
-          value={value}
-          onChange={this.handleOnChange}
-          placeholder={placeholder}
-          onFocus={onFocus}
-          onBlur={onBlur}
-          aria-label={intl.formatMessage({ id: 'Terra.datePicker.date' })}
-        />
+      <div
+        {...customProps}
+        className={dateInputContainerClasses}
+      >
+        <div className={dateInputClasses}>
+          <input
+            // Create a hidden input for storing the name and value attributes to use when submitting the form.
+            // The data stored in the value attribute will be the visible date in the date input but in ISO 8601 format.
+            data-terra-date-input-hidden
+            type="hidden"
+            name={name}
+            value={dateValue}
+          />
+          <Input
+            {...additionalInputProps}
+            className={cx('date-input-month')}
+            type="text"
+            name={'terra-date-month-'.concat(name)}
+            value={this.state.month}
```

Comment:
The date format for our supported locales can be in any of these formats:
`MM/DD/YYYY`
`DD/MM/YYYY`
`YYYY/MM/DD`

The ordering of these inputs would need to flex based on the locale's date format instead of always being MM/DD/YYYY

Reply:
Added: [847d4f0](https://github.com/cerner/terra-framework/pull/1004/commits/847d4f05fecc8074996a2d504e3cc216973b6eef). It's rather extensive considering the various changes down the line needed to support it, so I'm currently looking into moving some stuff into a separate file to clean things up a bit.

Reply:
+1 for composing each of these inputs in a separate function.

One option is we could add a new attribute (i.e. `Terra.datePicker.dateFormatOrder`)  in each of the translation files to indicate the ordering of the date format. For examples, for MM/DD/YYYY it would be set to 1, for DD/MM/YYYY -> 2, and YYYY/MM/DD -> 3.

It can then call these functions to compose the inputs in the order based on the new attribute.

Reply:
Updated: [452f10b](https://github.com/cerner/terra-framework/pull/1004/commits/452f10bd2fac12ea3c50aff9d7b90d0cff1d3b01). I rearranged things and composed the varying layouts better. I'm currently looking into doing this via the suggested attribute.

Reply:
Added in the attribute to make formatting decisions easier to reason about.